vavr/src/main/java/io/vavr/control/Either.java

```diff
@@ -163,6 +165,29 @@
     }
 
     /**
+     * Reduces many {@code Either}s into a single {@code Either} by transforming an
+     * {@code Iterable<Either<? extends L, ? extends R>>} into a {@code Either<? extends L, Seq<R>>}. If any of
+     * the {@code Either}s are {@link Either.Left}, then this returns a {@link Either.Left}.
+     *
+     * @param values An {@link Iterable} of {@code Either}s
+     * @param <L>    left type of the Eithers
+     * @param <R>    right type of the Eithers
+     * @return A {@code Either} of a {@link Seq} of results
+     * @throws NullPointerException if {@code values} is null
+     */
+    static <L,R> Either<L, Seq<R>> sequence(Iterable<? extends Either<? extends L, ? extends R>> values) {
+        Objects.requireNonNull(values, "values is null");
+        Vector<R> vector = Vector.empty();
```

Comment:
One thing is for sure, these implementations are totally non-fp like (not using flatmap, filter and so on). So i guess elegancy shouldn't be a concern but only performance. I am not familiar with the system vavr has for benchmarks, hopefully it's not too difficult to measure? The other thing is that clearly memory use will be higher for create+convert compared to only create (spike at ~2x the memory use I guess).